server/athenian/api/controllers/miners/github/pull_request.py

```diff
@@ -233,22 +233,21 @@ def __iter__(self) -> Generator[MinedPullRequest, None, None]:
                 if index is not None:
                     df.index = index
 
-    def drop(self, node_ids: Collection[int]) -> pd.Index:
+    def drop(self, node_ids: Collection[PullRequestID]) -> Collection[PullRequestID]:
         """
-        Remove PRs from the given collection of PR node IDs in-place.
+        Remove PRs by ID from the built collections.
 
-        Node IDs don't have to be all present.
+        PR IDs don't have to be all present.
 
-        :return: Actually removed node IDs.
+        :return: Actually removed PR IDs.
         """
-        removed = self._dfs.prs.index.get_level_values(0).intersection(node_ids)
-        if removed.empty:
-            return removed
-        self._dfs.prs.drop(removed, inplace=True)
-        for df in self._dfs.values():
-            df.drop(removed, inplace=True, errors="ignore",
-                    level=0 if isinstance(df.index, pd.MultiIndex) else None)
-        return removed
+        # remove just from the dataframes indexed on PullRequestID
+        toremove = self._dfs.prs.index.intersection(node_ids)
+        self._dfs.prs.drop(index=toremove, inplace=True)
```

Comment:
Performance: if `toremove` is empty, must return as-is. `drop` consolidates the blocks and is expensive. We can partially mitigate that by calling `prs.disable_consolidate()`, but only partially.